### PR TITLE
Move the stored namemap pre-population to namemap construction

### PR DESCRIPTION
**Pull request by levitte**
Prepopulation of the stored namemap from the legacy method object
database happened on first EVP fetch.  However, there are moments when
that prepopulation needs to happen even though no fetching has been
performed yet.  We therefore move pre-population to happen when the
namemap is constructed.

### CODE REVIEW

File: crypto/core_namemap.c [hunk 1]
```diff
@@ -335,3 +303,107 @@ int ossl_namemap_add_names(OSSL_NAMEMAP *namemap, int number,
 
     return number;
 }
+
+/*-
```
Comment (paulidale) [inline crypto/core_namemap.c]: `/*-` to open a multi-line comment?  I thought it was just `/*`

File: crypto/core_namemap.c [hunk 2]
```diff
@@ -335,3 +303,107 @@ int ossl_namemap_add_names(OSSL_NAMEMAP *namemap, int number,
...
+
+/* Creates an initial namemap with names found in the legacy method db */
+static void get_legacy_evp_names(const char *main_name, const char *alias,
+                                 void *arg)
+{
+    int main_id = ossl_namemap_add_name(arg, 0, main_name);
+
+    /*
+     * We could check that the returned value is the same as main_id,
+     * but since this is a void function, there's no sane way to report
```
Comment (paulidale) [inline crypto/core_namemap.c]: This is an internal function, it need not be `void`.  The calling functions are but isn't it early enough to change their behaviour?  [ or does it cascade up the call chain ]

File: crypto/core_namemap.c [hunk 3]
```diff
@@ -335,3 +303,107 @@ int ossl_namemap_add_names(OSSL_NAMEMAP *namemap, int number,
...
+{
+    OSSL_NAMEMAP *namemap =
+        openssl_ctx_get_data(libctx, OPENSSL_CTX_NAMEMAP_INDEX,
+                             &stored_namemap_method);
+
+#ifndef FIPS_MODE
+    if (namemap != NULL && ossl_namemap_empty(namemap)) {
+        /* Before pilfering, we make sure the legacy database is populated */
+        OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_CIPHERS
+                            |OPENSSL_INIT_ADD_ALL_DIGESTS, NULL);
```
Comment (paulidale) [inline crypto/core_namemap.c]: Space after `|`.

**Review by paulidale — COMMENTED** [review]
Superficially it looks good.  I need another pass over when I'm less tired.

File: crypto/core_namemap.c (hunk 1 above)
Comment (levitte) [reply crypto/core_namemap.c]: The dash is an accepted character to tell the 'indent' program not to reformat the comment.

Comment (paulidale) [reply crypto/core_namemap.c]: Cool.

File: crypto/core_namemap.c (hunk 2 above)
Comment (levitte) [reply crypto/core_namemap.c]: This is downstream from a callback for OBJ_NAME_do_all(), so the returned value is ignored either way...

Comment (paulidale) [reply crypto/core_namemap.c]: I suspected as much.  Thanks for explaining.

**Review by mattcaswell — APPROVED** [review]
LGTM assuming @pauliidale's nit is fixed

File: crypto/core_namemap.c (hunk 3 above)
Comment (levitte) [reply crypto/core_namemap.c]: Fixed

**Comment by levitte** [comment]
Merged.

437ad983c6ca254cd1d4367fb1d68338c82e8ee3 Move the stored namemap pre-population to namemap construction

**Pull request closed without merging**